src/test/java/com/checkmarx/flow/cucumber/integration/pullrequest/updatecomments/UpdatePullRequestCommentsSteps.java

```diff
@@ -94,20 +102,76 @@ private void initSca() {
         scaProperties.setAccessControlUrl("https://platform.checkmarx.net");
     }
 
+    private class ScanResultsAnswerer implements Answer<ScanResults> {
+
+        private FilterConfiguration filterConfiguration;
+
+        private FilterConfiguration switchFilterConfiguration(){
+            return filterConfiguration == filterMedium ? filterLow : filterMedium ;
+        }
+
+        @Override
+        public ScanResults answer(InvocationOnMock invocation) throws CheckmarxException {
+
+            filterConfiguration = switchFilterConfiguration();
+            ScanResults results = cxClientMock.getReportContent(sastFile, filterConfiguration);
+            results.setScanSummary(new CxScanSummary());
+
+            results.setAdditionalDetails(Collections.singletonMap(Constants.SUMMARY_KEY, new HashMap<>()));
+
+            return results;
+        }
+    }
+
+
     @Before
-    public void initMocks() {
+    public void initMocks(){
         initGitHubProperties();
         initSca();
         flowProperties.getBranches().add("udi-tests-2");
         flowProperties.setEnabledVulnerabilityScanners(Arrays.asList("sast"));
+        cxProperties.setOffline(true);
         initGitHubControllerSpy();
         initHelperServiceMock();
         setBranches();
+        initCxClientMock();
+    }
+
+    private void initCxClientMock() {
+        try {
+            ScanResultsAnswerer answerer = new ScanResultsAnswerer();
+            when(cxClientMock.getReportContentByScanId(anyInt(), any())).thenAnswer(answerer);
```

Comment:
'answerer' can be renamed into something more informative to make a sentence, e.g. 'withEmptySummary'.